llvm/include/llvm/Support/CommandLine.h

```diff
@@ -654,7 +655,7 @@ struct OptionValue<cl::boolOrDefault> final
 };
 
 template <>
-struct OptionValue<std::string> final : OptionValueCopy<std::string> {
+struct LLVM_ABI OptionValue<std::string> final : OptionValueCopy<std::string> {
```

Comment:
I think there are only extern template declarations for opt<> specializations, those also the ones that sufferer from MSVC bug i mentioned before where members are not getting exported from a dllexport'ed explicit template definition most likely because there deriving from built-in types like int.